axelrod/strategies/titfortat.py

```diff
@@ -438,6 +447,68 @@ def strategy(self, opponent: Player) -> Action:
 
         return C
 
+class Gradual(Player):
+    """
+    Similar to OriginalGradual, this is a player that punishes defections with a
+    growing number of defections but after punishing for `punishment_limit`
+    number of times enters a calming state and cooperates no matter what the
+    opponent does for two rounds.
+
+    This version of Gradual is an update of `OriginalGradual` and the difference
+    is that the `punishment_limit` is incremented whenever the opponent defects
+    (regardless of the state of the player).
+
+    Note that this version of `Gradual` appears in [CRISTAL-SMAC2018]_ however
+    this version of
+    `Gradual` does not give the results reported in [Beaufils1997]_ which is the
+    paper that first introduced the strategy. For a longer discussion of this
+    see: https://github.com/Axelrod-Python/Axelrod/issues/1294.
+
+    This version is based on  https://github.com/cristal-smac/ipd/blob/master/src/strategies.py#L224
+
+    Names:
+
+    - Gradual: [CRISTAL-SMAC2018]_
+    """
+
+    name = "Gradual"
+    classifier = {
+        "memory_depth": float("inf"),
+        "stochastic": False,
+        "makes_use_of": set(),
+        "long_run_time": False,
+        "inspects_source": False,
+        "manipulates_source": False,
+        "manipulates_state": False,
+    }
+
+    def __init__(self) -> None:
+
+        super().__init__()
+        self.calm_count = 0
+        self.punish_count = 0
+
+    def strategy(self, opponent: Player) -> Action:
+
+        if len(self.history) == 0:
+            return C
+
```

Comment:
niggle: too many empty lines (black might need running here)

Reply:
b7ff5c6 ran black on the file